packages/cache/src/cache.ts

```diff
@@ -185,14 +175,32 @@ export async function saveCache(
     const fileSizeLimit = 10 * 1024 * 1024 * 1024 // 10GB per repo limit
     const archiveFileSize = utils.getArchiveFileSizeInBytes(archivePath)
     core.debug(`File Size: ${archiveFileSize}`)
-    if (archiveFileSize > fileSizeLimit) {
+    if (archiveFileSize > fileSizeLimit && !utils.isGhes()) {
       throw new Error(
         `Cache size of ~${Math.round(
           archiveFileSize / (1024 * 1024)
         )} MB (${archiveFileSize} B) is over the 10GB limit, not saving cache.`
       )
     }
 
+    const cacheSize = utils.isGhes() ? archiveFileSize : undefined
+    core.debug('Reserving Cache')
+    cacheId = await cacheHttpClient.reserveCache(key, paths, {
+      compressionMethod,
+      cacheSize
+    })
+    if (cacheId === -1) {
+      throw new ReserveCacheError(
+        `Unable to reserve cache with key ${key}, another job may be creating this cache.`
+      )
+    }
+    if (cacheId === -2) {
+      throw new ReserveCacheError(
+        `Cache size of ~${Math.round(archiveFileSize / (1024 * 1024))} MB (${archiveFileSize} B) is over the data cap limit, not saving cache.`
```

Comment:
Can we use the API response error message directly. It has the cap mentioned which would be a valuable info in action log.

Reply:
Also, this will allow multiple input validations to show appropriate error message.

Reply:
Done